### PR TITLE
Introduce go build of functests

### PR DESCRIPTION
**Pull request by 0xFelix**
**What this PR does / why we need it**:

This introduces a go build of the functests without bazel. The existing make target 'build-functest' was renamed to 'bazel-build-functest' and aliased with 'build-functest'.

This allows building the functests without bazel as part of `make go-build` or with an additional target `make go-build-functests`. It respects build tags set with `KUBEVIRT_GO_BUILD_TAGS`.

Previously the build tags were not respected and there was no separate build target for building only the functional tests with go.

**Release note**:

```release-note
None
```

### CODE REVIEW

File: hack/build-go.sh [hunk 1]
```diff
@@ -23,6 +23,8 @@ source hack/common.sh
 source hack/config.sh
 source hack/version.sh
 
+source hack/go-build-functests.sh
```
Comment (rmohr) [inline hack/build-go.sh:26]: Is sourcing here a left-over or do you want to build in addition to all go binaries too?

Comment (0xFelix) [reply hack/build-go.sh:26]: This is to keep current behavior.

See: https://github.com/kubevirt/kubevirt/blob/8b7377e297436489f021c021e473d03997f5d5d9/hack/build-go.sh#L147

Comment (rmohr) [reply hack/build-go.sh:26]: I see. So you want to have a break-out target but still build it together with the rest as well.

Comment (0xFelix) [reply hack/build-go.sh:26]: Exactly!

**Comment by EdDev** [comment]
> This introducea a go build of the functests without bazel. The existing make target 'build-functest' was renamed to 'bazel-build-functest' and aliased with 'build-functest'.

Could you please elaborate on the reasoning behind this change?

**Comment by 0xFelix** [comment]
@EdDev We already have a go-build for the KubeVirt binaries which is used downstream. The `go-build-functests` helps to keep downstream builds in sync e.g. it allows to set build tags, which cannot bet set in functest builds at the moment.

**Comment by EdDev** [comment]
> @EdDev We already have a go-build for the KubeVirt binaries which is used downstream. The `go-build-functests` helps to keep downstream builds in sync e.g. it allows to set build tags, which cannot bet set in functest builds at the moment.

Unfortunately, I have no context here and have hard time understanding what this means.
If this is meant only for experts that are familiar with this subject, then fine.

But it will be great to see an explanation for others who are less familiar, at least to understand at high level what is the current problem and what it solves.

**Comment by rmohr** [comment]
Adjusting the PR description a little bit to better outline that you just introduce an extra buil-only-functest-with-go target while still building it also as part of the normal go build may make sense. Like @EdDev suggested.

**Comment by 0xFelix** [comment]
> But it will be great to see an explanation for others who are less familiar, at least to understand at high level what is the current problem and what it solves.

It allows building the functests without bazel as part of `make go-build` or with an additional target `make go-build-functests`. It respects build tags set with `KUBEVIRT_GO_BUILD_TAGS`.

Previously the build tags were not respected and there was no separate build target for building only the functional tests with go.

**Comment by 0xFelix** [comment]
This is useful in combination with this PR: https://github.com/kubevirt/kubevirt/pull/8955

**Pull request merged**